### PR TITLE
Fix gateway status

### PR DESCRIPTION
**Pull request by ryaplots**
#### Summary

This PR fixes the Gateway status being not visible in the Gateways list.

#### Changes

- Removed the getHostFromUrl()

#### Testing

Open gateways list

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Comment by adriansmares** [comment]
Does everything still work if you edit the gateway server address to `localhost:1234` ? That was the use case here

**Comment by ryaplots** [comment]
> Does everything still work if you edit the gateway server address to `localhost:1234` ? That was the use case here

If I do that the status of all gateways appears as "Other cluster"

**Comment by adriansmares** [comment]
The problem is that `getHostFromUrl` won't correctly parse `gateway.gateway_server_address` since it is not really a fully fledged URL. I think the correct solution here is to adapt `getHostFromUrl` to match the logic that we use in the CLI:

https://github.com/TheThingsNetwork/lorawan-stack/blob/911078b6cde10c3d2c319efcd3e90e47355332aa/cmd/ttn-lw-cli/commands/utils.go#L31-L45

Basically `new URL` should be used only if a scheme is present (that's what the `://` check does). Note that splitting between host and port should be handled with care as you cannot just use `:` since `:` appears in IPv6 addresses. I'm not sure what the correct equivalent for `net.SplitHostPort` is in JavaScript.

**Comment by ryaplots** [comment]
> The problem is that `getHostFromUrl` won't correctly parse `gateway.gateway_server_address` since it is not really a fully fledged URL. I think the correct solution here is to adapt `getHostFromUrl` to match the logic that we use in the CLI:
> 
> https://github.com/TheThingsNetwork/lorawan-stack/blob/911078b6cde10c3d2c319efcd3e90e47355332aa/cmd/ttn-lw-cli/commands/utils.go#L31-L45

You are right indeed. I removed it because I remember there was a PR that added the `getHostFromUrl` in a lot of places where it wasn't before. So I presumed this was one of those places and it does not work here. But indeed extending it would make much more sense. I am on it

**Comment by ryaplots** [comment]
@adriansmares I did some research and couldn't find an equivalent to `net.SplitHostPort` besides handling the actual `location` or using `new URL()`. So what I did was match the string to a `regex` to handle IPv6 addresses. And after that if the url contains a `:` I can add `://` before and then `new URL()` does not return undefined anymore. I tested this with URLs like 'tti.staging1.cloud.thethings.industries', 'localhost', 'localhost:1234' and 'https://example' and it worked.

File: pkg/webui/lib/host-from-url.js [hunk 1]
```diff
@@ -19,8 +19,19 @@
...
+    if (url.match(IPv6RegexExp)) {
+      return url
+    } else if (url.includes('://')) {
+      return new URL(url).hostname
+    } else if (url.includes(':')) {
+      return new URL(`https://${url}`).hostname
+    }
+
+    return url
   } catch (error) {
```
Comment (kschiffer) [inline pkg/webui/lib/host-from-url.js]:
```suggestion
  try {
    if (url.matches(/^[a-zA-Z0-9]+:\/\/.*/)) {
      return new URL(url).hostname
    }
    return new URL(`http://${url}`).hostname
  } catch (error) {
    return url
  }
```

Let's simplify this.

**Review by kschiffer — CHANGES_REQUESTED** [review]

File: pkg/webui/lib/host-from-url.js [hunk 2]
```diff
@@ -20,7 +20,11 @@
...
 export default url => {
   try {
-    return new URL(url).hostname
+    if (url.match(/^[a-zA-Z0-9]+:\/\/.*/)) {
+      return new URL(url).hostname
+    }
+
+    return new URL(`http://${url}`).hostname
   } catch (error) {
     return undefined
```
Comment (kschiffer) [inline pkg/webui/lib/host-from-url.js]:
```suggestion
    return url
```

I think that's more resilient.

**Review by kschiffer — APPROVED** [review]

**Review by adriansmares — APPROVED** [review]
I'll backport this one into `rc1` everywhere.

**Pull request merged**